packages/core/src/Stores/notification-store.js

```diff
@@ -245,14 +252,23 @@ export default class NotificationStore extends BaseStore {
                     ASK_UK_FUNDS_PROTECTION,
                 } = cashier_validation ? getCashierValidations(cashier_validation) : {};
 
-                this.addVerificationNotifications(identity, document);
                 const needs_poa =
                     is_10k_withdrawal_limit_reached &&
                     (needs_verification.includes('document') || document?.status !== 'verified');
                 const needs_poi = is_10k_withdrawal_limit_reached && identity?.status !== 'verified';
+                const onfido_submissions_left = identity.services.onfido.submissions_left;
+
+                this.addVerificationNotifications(identity, document);
 
                 if (needs_poa) this.addNotificationMessage(this.client_notifications.needs_poa);
                 if (needs_poi) this.addNotificationMessage(this.client_notifications.needs_poi);
+                if (poi_name_mismatch && identity.services.onfido.last_rejected) {
```

Comment:
Do we need to add optional chaining for `identity` variable?

Reply:
By right, we shouldn't since the code is executed for logged in users only and BE should provide the object for them, but I added the check to be safe